### PR TITLE
Refine Tibia Asset Integration with Port and Adapter Pattern

### PR DESCRIPTION
**Pull request by lgrossi**
## Overview
This Merge Request refines the integration of Tibia asset data into the Ryot system by implementing a clear port and adapter pattern by establishing a new dedicated crate, `ryot_tibia_assets`. The refactor involves structuring our system to handle Tibia's assets through well-defined interfaces (ports) and their implementations (adapters). This architecture replaces the previous mixed implementation, enhancing modularity and maintainability.

## Changes
- **Core Definitions**: Updated `VisualElements` and `VisualElement` structures within `ryot_assets/src/core/visual_elements.rs` to represent visual elements within Ryot in an agnostic way.
- **FrameGroup Enhancements**: Modified the `FrameGroup` enum to provide utility methods and clearer categorization, improving readability and usability.
- **Animation and Properties**: Streamlined handling of animation sequences and item properties, ensuring consistency and maintainability across our asset handling.
- **Category Handling**: Refined category determination logic to accommodate custom categorizations directly within the asset definitions, allowing for flexible asset categorization based on dynamic data.
- **Conversion Crate**: Introduces `ryot_tibia_assets` crate to facilitate robust and error-resistant data conversion from Tibia's formats to Ryot's internal structures.
- **Interface Implementation**: Consolidated data handling through a new interface in `content_type.rs`, abstracting sprite sheet data handling into a more manageable format.

## Motivation
The motivation for this refactor was to eliminate the direct and mixed handling of external Tibia data within the Ryot system's core logic. By defining clear interfaces and implementing them through adapters, we align our architecture with Domain-Driven Design principles, significantly improving the system's resilience and scalability.

## Dependencies
Simplified ryot crates dependency from assets crate, depending solely on the `ryot_core` crate. We inverted the dependency by making `ryot_grid` depend on `ryot_assets` instead.

## Impact
This architectural enhancement decouples external asset data handling from the Ryot system's core logic, facilitating easier updates, better testability, and more robust data integration. It sets a solid foundation for future enhancements and potential integration with other external data sources.

### CODE REVIEW

File: crates/ryot/Cargo.toml [hunk 1]
```diff
@@ -19,8 +19,7 @@ serde_repr = "0.1"
 ryot_assets = { path = "../ryot_assets", version = "0.2.0" }
```
Comment (luan) [inline crates/ryot/Cargo.toml:19]: is `version =` necessary when doing paths?

File: crates/ryot_assets/src/core/tests.rs [hunk 2]
```diff
@@ -113,26 +114,11 @@ fn sprite_sheet_set_fixture() -> SpriteSheetDataSet {
         }),
     ];
 
-    SpriteSheetDataSet::from_content(&vec)
+    (&*vec).into()
```
Comment (luan) [inline crates/ryot_assets/src/core/tests.rs:117]: ew

**Review by luan — APPROVED** [review]

File: crates/ryot/Cargo.toml (hunk 1 above)
Comment (lgrossi) [reply crates/ryot/Cargo.toml:19]: It's not needed, but it doesn't hurt keeping it, it also helps guaranteeing we are always release the same version to all of them.

File: crates/ryot_assets/src/core/tests.rs (hunk 2 above)
Comment (lgrossi) [reply crates/ryot_assets/src/core/tests.rs]: okok, I've removed all the usages of &* from the codebase hahaha

**Pull request merged**